app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -26,11 +32,21 @@ export type Props = {
   fetching: boolean;
   semesters: Array<CompanySemesterEntity>;
   push: (arg0: string) => void;
-  selectedOption: Option;
+  selectedSemesterOption: SemesterOption;
+  selectedEventOption: EventOption;
   router: any;
+  authToken: string;
 };
 type State = {
   clickedCompanyInterest: number;
+  generatedCSV:
+    | {
+        url: string;
+        filename: string;
+      }
+    | null
+    | undefined;
```

Comment:
Nice work! Use `generatedCSV?: { url: string; filename: string; }` whenever a field may be undefined^